compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/InitMethodGen.java

```diff
@@ -267,41 +277,57 @@ public void enrichPkgWithInitializers(Map<String, BIRFunctionWrapper> birFunctio
     private BIRNode.BIRFunction generateExecuteFunction(BIRNode.BIRPackage pkg, boolean serviceEPAvailable,
                                                         BIRNode.BIRFunction mainFunc,
                                                         BIRNode.BIRFunction testExecuteFunc, String typeOwnerClass) {
-        BIRNode.BIRVariableDcl retVar = new BIRNode.BIRVariableDcl(null, errorOrNilType, new Name("%ret"),
-                VarScope.FUNCTION, VarKind.RETURN, "");
+        BIRNode.BIRVariableDcl retVar = new BIRNode.BIRVariableDcl(null, errorOrNilType,
+                new Name("%ret"), VarScope.FUNCTION, VarKind.RETURN, "");
         BIROperand retVarRef = new BIROperand(retVar);
         List<BIROperand> functionArgs = new ArrayList<>();
         BInvokableType funcType = new BInvokableType(Collections.emptyList(), null, errorOrNilType, null);
-        BIRNode.BIRFunction modExecFunc = new BIRNode.BIRFunction(null, new Name(MODULE_EXECUTE_METHOD), 0,
-                funcType, null, 0, VIRTUAL);
+        BIRNode.BIRFunction modExecFunc = new BIRNode.BIRFunction(null, new Name(MODULE_EXECUTE_METHOD),
+                0, funcType, null, 0, VIRTUAL);
         List<BType> paramTypes = new ArrayList<>();
         List<BIRNode.BIRFunctionParameter> parameters = new ArrayList<>();
         List<BIRNode.BIRParameter> requiredParameters = new ArrayList<>();
         int argsCount = 0;
+        BIRNode.BIRFunctionParameter cliArgVar = null;
 
         if (mainFunc != null) {
-            paramTypes = mainFunc.type.paramTypes;
-            parameters = mainFunc.parameters;
-            requiredParameters = mainFunc.requiredParams;
-            argsCount += modExecFunc.parameters.size();
-        } else if (testExecuteFunc != null) {
+            List<BIRNode.BIRFunctionParameter> mainParameters = mainFunc.parameters;
+            Name argName = new Name("%_cli");
+            cliArgVar = new BIRNode.BIRFunctionParameter(null, symbolTable.anyType, argName, VarScope.FUNCTION,
+                    VarKind.ARG, "", false, false);
```

Comment:
Is it okay passing `""` to metaVarName? I see below also we are passing the same? Won't having the same affect the correct behaviour? What happens if we pass `null`?

Reply:
the `metaVarName` is used only in the `FPCall` terminator generation. And it has no effect in other places. Therefore, having either `""` value or `null` will have the same effect. I changed it to pass null.